include/ibm/management_console_rest.hpp

```diff
@@ -924,6 +924,45 @@ void handleCsrRequest(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
         "SignCSR", csrString);
 }
 
+inline void
+    handlePassThrough(const crow::Request& req,
+                      const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                      const std::string& name)
+{
+    std::vector<int32_t> command;
+
+    if (!redfish::json_util::readJson(req, asyncResp->res, "Send", command))
+    {
+        BMCWEB_LOG_DEBUG << "Not a Valid JSON";
+        asyncResp->res.result(boost::beast::http::status::bad_request);
+        return;
+    }
+
+    if (command.size() == 0)
+    {
+        BMCWEB_LOG_ERROR << "Command is empty";
+        asyncResp->res.result(boost::beast::http::status::bad_request);
+        return;
+    }
+
+    auto respHandler = [asyncResp](const boost::system::error_code ec) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "handlePassThrough respHandler got error, ec = "
+                             << ec.value();
+            asyncResp->res.result(
+                boost::beast::http::status::internal_server_error);
+            return;
+        }
+    };
+
+    crow::connections::systemBus->async_method_call(
+        respHandler, "org.open_power.OCC.Control",
+        "/org/open_power/control/" + name, "org.open_power.OCC.PassThrough",
+        "Send", command);
```

Comment:
Looking at the interface, Looks like it has response data.
https://github.com/openbmc/phosphor-dbus-interfaces/blob/master/yaml/org/open_power/OCC/PassThrough.interface.yaml#L12
@cjcain Do we need to do anything there? If so, can you get @lxwinspur a command to test with 
How do we get that out? 

An example command is 

busctl call org.open_power.OCC.Control /org/open_power/control/occ0 org.open_power.OCC.PassThrough Send ai 4 0 0 1 32

ai 770 49 0 0 2 253 147 0 11 0 3 12 0 0 0 0 0 0 0 0 0 0 111 112 95 112 49 48 95 50 49 48 55 50 50 97 0 0 83 69 78 83 79 82 5 1 84 69 77 80 0 16 8 24 0 0 0 255 0 37 85 0 0 0 0 255 0 36 85 0 0 0 0 255 0 38 85 0 0 0 0 255 0 38 85 0 0 0 0 255 0 36 85 0 0 0 0 255 0 36 85 0 0 0 0 255 0 37 85 0 0 0 0 255 0 37 85 0 0 0 0 255 0 38 85 0 0 0 0 255 0 37 85 0 0 0 0 255 0 36 85 0 0 0 0 255 0 36 85 0 0 0 0 255 0 37 85 0 0 0 0 255 0 36 85 0 0 0 0 255 0 38 85 0 0 0 0 255 0 38 85 0 0 0 0 255 255 0 0 0 0 0 0 255 2 37 76 0 0 0 0 255 7 39 72 0 0 0 0 255 255 0 0 0 0 0 0 255 2 35 76 0 0 0 0 255 7 37 72 0 0 0 0 255 6 37 85 0 0 0 0 0 9 41 95 0 70 82 69 81 0 2 6 16 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 80 79 87 82 0 2 22 13 0 0 0 0 5 2 0 0 0 12 21 25 0 0 0 0 2 231 108 187 0 61 0 0 0 0 6 3 0 0 0 12 21 25 0 0 0 0 1 172 46 36 0 35 0 0 0 0 9 4 0 0 0 12 21 25 0 0 0 0 3 161 243 22 0 77 0 0 0 0 10 5 0 0 0 12 21 25 0 0 0 0 1 206 57 213 0 38 0 0 0 0 1 6 0 0 0 12 21 25 0 0 0 0 4 57 87 162 0 89 0 0 0 0 2 7 0 0 0 12 21 25 0 0 0 0 2 93 239 60 0 50 0 0 0 0 29 8 0 0 0 12 21 25 0 0 0 0 1 9 209 65 0 22 0 0 0 0 0 9 0 0 0 12 21 25 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 10 0 0 0 12 21 25 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 11 0 0 0 12 21 25 0 0 0 0 0 0 0 0 0 0 0 0 0 0 16 13 0 0 0 12 21 25 0 0 0 0 17 165 56 191 1 119 0 0 0 0 18 14 0 0 0 12 21 25 0 0 0 0 1 105 109 143 0 30 0 0 0 0 13 15 0 0 0 12 21 25 0 0 0 0 3 196 1 41 0 79 67 65 80 83 0 3 15 1 8 192 3 88 7 8 8 192 6 64 1 244 0 0 0 69 88 84 78 0 1 12 8 70 77 73 78 0 0 129 7 209 0 0 0 70 68 73 83 0 0 82 10 240 0 0 0 70 66 65 83 0 0 64 12 28 0 0 0 70 85 84 0 0 0 58 12 128 0 0 0 70 77 65 88 0 0 58 12 128 0 0 0 67 76 73 80 0 0 58 0 0 0 0 0 87 79 70 67 0 0 22 0 0 255 255 0 69 82 82 72 0 0 17 1 25 255 34 25


curl -k  -X POST
https://${bmc}/ibm/v1/OCC/Control/occ0/Actions/PassThrough.Send -d '{"Send": [4, 0, 0, 1, 32]}'

Reply:
Yes, response data is required.  The following busctrl command (OCC Poll command) has response data:
Command:
`busctl call org.open_power.OCC.Control /org/open_power/control/occ0 org.open_power.OCC.PassThrough Send ai 4 0 0 1 32`
Response:
`ai 744 28 0 0 2 227 147 0 15 0 3 12 0 0 0 0 0 0 0 0 0 0 111 112 95 112 49 48 95 50 49 48 56 49 51 97 0 0 83 69 78 83 79 82 5 1 84 69 77 80 0 16 8 18 0 0 0 255 0 33 85 0 0 0 0 255 0 33 85 0 0 0 0 255 0 31 85 0 0 0 0 255 0 32 85 0 0 0 0 255 0 32 85 0 0 0 0 255 0 32 85 0 0 0 0 255 0 31 85 0 0 0 0 255 0 32 85 0 0 0 0 255 0 31 85 0 0 0 0 255 0 31 85 0 0 0 0 255 0 33 85 0 0 0 0 255 0 33 85 0 0 0 0 255 0 33 85 0 0 0 0 255 0 33 85 0 0 0 0 255 0 32 85 0 0 0 0 255 0 31 85 0 0 0 0 255 6 24 85 0 0 0 0 0 9 32 95 0 70 82 69 81 0 2 6 16 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 0 0 0 0 14 212 80 79 87 82 0 2 22 14 0 0 0 0 30 0 0 0 0 5 243 38 0 0 0 0 4 70 194 142 0 184 0 0 0 0 5 2 0 0 0 5 243 38 0 0 0 0 1 220 171 65 0 82 0 0 0 0 6 3 0 0 0 5 243 38 0 0 0 0 1 209 59 138 0 78 0 0 0 0 9 4 0 0 0 5 243 38 0 0 0 0 1 94 228 157 0 59 0 0 0 0 10 5 0 0 0 5 243 38 0 0 0 0 1 102 206 185 0 60 0 0 0 0 1 6 0 0 0 5 243 38 0 0 0 0 0 71 53 218 0 12 0 0 0 0 2 7 0 0 0 5 243 38 0 0 0 0 0 118 250 219 0 20 0 0 0 0 29 8 0 0 0 5 243 38 0 0 0 0 0 126 87 57 0 22 0 0 0 0 0 9 0 0 0 5 243 38 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 10 0 0 0 5 243 38 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 11 0 0 0 5 243 38 0 0 0 0 0 0 0 0 0 0 0 0 0 0 16 13 0 0 0 5 243 38 0 0 0 0 4 99 52 109 0 200 0 0 0 0 18 14 0 0 0 5 243 38 0 0 0 0 0 160 206 62 0 27 0 0 0 0 13 15 0 0 0 5 243 38 0 0 0 0 1 217 198 229 0 79 67 65 80 83 0 3 15 1 8 192 3 54 7 8 8 192 6 64 1 244 0 0 0 69 88 84 78 0 1 12 8 70 77 73 78 0 0 129 7 209 0 0 0 70 68 73 83 0 0 82 10 240 0 0 0 70 66 65 83 0 0 64 12 28 0 0 0 70 85 84 0 0 0 22 14 216 0 0 0 70 77 65 88 0 0 22 14 216 0 0 0 67 76 73 80 0 0 22 0 0 0 0 0 87 79 70 67 0 0 22 0 0 255 255 0 69 82 82 72 0 0 17 25 0 0 0 0`

Reply:
Ack.
Updated, But I can not test it, Could you test it?

Reply:
> But I can not test it, Could you test it? 

Why can't you test it ?

Reply:
@patelabhishek9893 You should get something back. Do you not?

Reply:
@lxwinspur could you look into this?

Reply:
> @lxwinspur could you look into this?

Yes, I tested it and passes

Reply:
I test on rain104bmc system

root@rain104bmc:~# busctl call org.open_power.OCC.Control /org/open_power/control/occ0 org.open_power.OCC.PassThrough Send ai 4 0 0 1 32
ai 792 17 0 0 3 19 147 0 15 0 3 12 0 0 0 0 0 0 0 0 0 0 111 112 95 112 49 48 95 50 49 48 55 50 49 97 0 0 83 69 78 83 79 82 5 1 84 69 77 80 0 16 8 24 0 0 0 255                                 0 39 85 0 0 0 0 255 0 39 85 0 0 0 0 255 0 38 85 0 0 0 0 255 0 39 85 0 0 0 0 255 0 37 85 0 0 0 0 255 0 37 85 0 0 0 0 255 0 37 85 0 0 0 0 255 0 37 85 0 0 0 0                                 255 0 36 85 0 0 0 0 255 0 37 85 0 0 0 0 255 0 38 85 0 0 0 0 255 0 39 85 0 0 0 0 255 0 38 85 0 0 0 0 255 0 38 85 0 0 0 0 255 0 38 85 0 0 0 0 255 0 38 85 0 0 0                                 0 255 255 0 0 0 0 0 0 255 3 27 65 0 0 0 0 255 7 29 72 0 0 0 0 255 255 0 0 0 0 0 0 255 3 26 65 0 0 0 0 255 7 29 72 0 0 0 0 255 6 31 85 0 0 0 0 0 9 40 95 0 70                                 82 69 81 0 2 6 16 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0                                 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 0 0 0 0 12 124 80 79 87 82 0 2 22 14 0 0 0 0 30 0 0 0 1 66 20 199 0 0                                 0 0 231 126 236 147 0 184 0 0 0 0 5 2 0 0 1 66 20 199 0 0 0 0 91 213 47 239 0 75 0 0 0 0 6 3 0 0 1 66 20 199 0 0 0 0 87 239 197 165 0 67 0 0 0 0 9 4 0 0 1 66                                 20 199 0 0 0 0 82 177 37 121 0 66 0 0 0 0 10 5 0 0 1 66 20 199 0 0 0 0 81 108 195 116 0 65 0 0 0 0 1 6 0 0 1 66 20 199 0 0 0 0 21 102 240 76 0 17 0 0 0 0 2                                 7 0 0 1 66 20 199 0 0 0 0 25 20 74 185 0 20 0 0 0 0 29 8 0 0 1 66 20 199 0 0 0 0 25 191 159 20 0 21 0 0 0 0 0 9 0 0 1 66 20 199 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0                                 10 0 0 1 66 20 199 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 11 0 0 1 66 20 199 0 0 0 0 0 0 0 0 0 0 0 0 0 0 16 13 0 0 1 66 20 199 0 0 0 0 236 143 156 200 0 184 0 0 0 0                                 18 14 0 0 1 66 20 199 0 0 0 0 31 207 60 104 0 25 0 0 0 0 13 15 0 0 1 66 20 199 0 0 0 0 94 28 166 140 0 74 67 65 80 83 0 3 15 1 8 192 3 31 7 8 8 192 6 64 2 18                                8 0 0 0 69 88 84 78 0 1 12 8 70 77 73 78 0 0 129 7 209 0 0 0 70 68 73 83 0 0 82 10 240 0 0 0 70 66 65 83 0 0 64 12 28 0 0 0 70 85 84 0 0 0 58 12 128 0 0 0 70                                 77 65 88 0 0 58 12 128 0 0 0 67 76 73 80 0 0 58 0 0 0 0 0 87 79 70 67 0 0 22 0 0 255 255 0 69 82 82 72 0 0 32 5 0 0 0 0

where I run the below command I get Internal Server Error

export bmc=rain104bmc.aus.stglabs.ibm.com:18080
export bmc_token=`curl -k -H "Content-Type: application/json" -X POST https://${bmc}/login -d '{"username" :  "root", "password" :  "0penBmc"}' | grep token | awk '{print $2;}' | tr -d '"'`
curl -k -H "X-Auth-Token: $bmc_token" -X POST https://${bmc}/ibm/v1/OCC/Control/occ0/Actions/PassThrough.Send -d '{"Send": [4,0,0,1,32]}'

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    84  100    37  100    47    234    297 --:--:-- --:--:-- --:--:--   531

Reply:
Please see: https://github.com/ibm-openbmc/bmcweb/commit/996e793eae1717e9088b03343f367b5b4d4d417a

Note: should '{"Send": [0,0,1,32]}', not '{"Send": [4,0,0,1,32]}'

And I tested it passes.
Could you share your error message here?

Reply:
@lxwinspur You saw data back like 
```
 792 17 0 0 3 19 147 0 15 0 3 12 0 0 0 0 0 0 0 0 0 0 111 112 95 112 49 48 95 50 49 48 55 50 49 97 0 0 83 69 78 83 79 82 5 1 84 69 77 80 0 16 8 24 0 
```

Reply:
Yes, I tested it on dubai system

Reply:
look like I am unable to find out why it giving me  Internal Server Error. it might be something else, and on my end it is hard to debug

code looks good. So if @lxwinspur said it tested and we like to merge it then we will go ahead and merge this code.